lib/dialects/mariadb/query.js

```diff
@@ -179,19 +179,21 @@ class Query extends AbstractQuery {
     if (!this.model || !this.model.fieldRawAttributesMap) {
       return;
     }
+    // Occurs when using mariadb driver with mysql
+    const wasParsed = rows.meta.some(m => m.type === 'JSON');
     for (const _field of Object.keys(this.model.fieldRawAttributesMap)) {
       const modelField = this.model.fieldRawAttributesMap[_field];
-      if (modelField.type instanceof DataTypes.JSON) {
+      const { type, fieldName } = modelField;
+      if (type instanceof DataTypes.JSON) {
         //value is return as String, no JSON
-        rows = rows.map(row => {
-          row[modelField.fieldName] = row[modelField.fieldName] ? JSON.parse(
-            row[modelField.fieldName]) : null;
+        for (const r of rows) {
+          if (!wasParsed) {
+            r[fieldName] = r[fieldName] ? JSON.parse(r[fieldName]) : null;
+          }
           if (DataTypes.JSON.parse) {
-            return DataTypes.JSON.parse(modelField, this.sequelize.options,
-              row[modelField.fieldName]);
+            r[fieldName] = DataTypes.JSON.parse(modelField, this.sequelize.options, r[fieldName]);
```

Comment:
This does not seem equivalent to the previous implementation, is this intentional? Previously the array element itself was replaced by the result of the `DataTypes.JSON.parse` call. Now the `r[fieldName]` is replaced, right?

Reply:
Seems that way, misinterpreted the code due to lack of docs, will revert that assignment.

Reply:
Unless the purpose of `DataTypes.JSON.parse` is to mutate new data by not returning, this doesn't really work anyways since `rows` is discarded.